lib/fragment.js

```diff
@@ -81,6 +82,13 @@ module.exports = class Fragment extends EventEmitter {
 
         this.insertStart();
 
+        let responseStream = response;
+        const contentEncoding = response.headers['content-encoding'];
+        if (contentEncoding &&
+            (contentEncoding === 'gzip' || contentEncoding === 'deflate')) {
+            responseStream = response.pipe(zlib.createUnzip());
```

Comment:
You've added one more stream but haven't added one more 'error' handler for it, it can lead to error that will be thrown.